src/js/components/RangeSelector/RangeSelector.js

```diff
@@ -118,141 +124,124 @@ class RangeSelector extends Component {
       }
     }
     if (nextValues) {
-      this.setState({ lastChange: changing, moveValue: value }, () => {
-        onChange(nextValues);
+      setLastState({
+        lastChange: changing,
+        lastValues: nextValues,
       });
+      setMoveValue(value);
     }
   };
 
-  mouseUp = () => {
-    this.setState({ changing: undefined });
-    window.removeEventListener('mousemove', this.mouseMove);
-    window.removeEventListener('mouseup', this.mouseUp);
+  const mouseUp = () => {
+    setChanging(undefined);
   };
 
-  render() {
-    const {
-      color,
-      direction,
-      forwardRef,
-      invert,
-      max,
-      messages,
-      min,
-      onChange,
-      opacity,
-      round,
-      size,
-      step,
-      values,
-      theme: propsTheme,
-      ...rest
-    } = this.props;
-    const theme = this.context || propsTheme;
-    const { nextLower, nextUpper } = this.state;
-
-    const lower = nextLower !== undefined ? nextLower : values[0];
-    const upper = nextUpper !== undefined ? nextUpper : values[1];
-    // It needs to be true when vertical, due to how browsers manage height
-    const fill = direction === 'vertical' ? true : 'vertical';
-
-    return (
-      <Container
-        ref={this.containerRef}
-        direction={direction === 'vertical' ? 'column' : 'row'}
+  useEffect(() => {
+    window.addEventListener('mousemove', mouseMove);
+    window.addEventListener('mouseup', mouseUp);
+
+    return () => {
+      window.removeEventListener('mousemove', mouseMove);
+      window.removeEventListener('mouseup', mouseUp);
+    };
+  }, [changing]);
+
+  return (
+    <Container
+      ref={containerRef}
+      direction={direction === 'vertical' ? 'column' : 'row'}
+      fill={fill}
+      {...rest}
+      onClick={onChange ? onClick : undefined}
+    >
+      <Box
+        style={{ flex: `${lower - min} 0 0` }}
+        background={
+          invert
+            ? // preserve existing dark, instead of using darknes
```

Comment:
This comment makes it a little hard to read it, what do you think about moving it?

Reply:
I think thats a good idea :+1: